### PR TITLE
JAMES-2598 Compile Cassandra+ES+Swift+RabbitMQ+Ldap product

### PR DESCRIPTION
**Pull request by hoangdat**
CI: https://issues.apache.org/jira/browse/JAMES-2598?jql=project%20%3D%20JAMES

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
`MailQueueContract.concurrentEnqueueDequeueShouldNotFail:387 » NotTerminated` again 

test this please

**Comment by hoangdat** [comment]
force push, rebase master

File: README.adoc [hunk 1]
```diff
@@ -195,13 +196,13 @@ If you are using a a fresh installation of Docker, your DOCKER_HOST should be un
 
 This feature is available for three configurations :
 
- * Java 8 + Guice + Cassandra + RabbitMQ + Swift + ElasticSearch
+ * Java 8 + Guice + Cassandra + RabbitMQ + Swift + ElasticSearch + Ldap
```
Comment (trantienduchn) [inline README.adoc]: we don't update `how to run james with ldap` in this section. See below products! Document for ldap is updated in `packaging` section.

Comment (hoangdat) [reply README.adoc]: Follow other product with Ldap, I removed it in this file.

**Comment by hoangdat** [comment]
rebase master, squash fixup

**Comment by trantienduchn** [comment]
```
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR] Errors: 
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR]   MailQueueContract.concurrentEnqueueDequeueShouldNotFail:387 » NotTerminated
[a25cf2e2ebdf190ddde777be051238429321930e] [INFO] 
```
test this please

**Comment by trantienduchn** [comment]
```
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR] Process Exit Code: 0
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR] Crashed tests:
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR] org.apache.james.mpt.smtp.RabbitMQSmtpStarttlsCommandTest
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR] Command was /bin/sh -c cd /james-project/mpt/impl/smtp/cassandra-rabbitmq-swift && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -Xms512m -Xmx1024m -jar /james-project/mpt/impl/smtp/cassandra-rabbitmq-swift/target/surefire/surefirebooter4797568518266916098.jar /james-project/mpt/impl/smtp/cassandra-rabbitmq-swift/target/surefire 2018-11-26T14-25-59_278-jvmRun1 surefire453687739757049941tmp surefire_1285315791604528618952tmp
```
I noticed there are sometimes smtp rabbitmq test crashed recently. test this please

**Comment by chibenwa** [comment]
```
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR]   SpamAssassinContract.expungingSpamMessageShouldNotImpactSpamAssassinState:493 » ConditionTimeout
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR]   SpamAssassinContract.spamShouldBeDeliveredInSpamMailboxWhenSameMessageHasAlreadyBeenMovedToSpam:146 » ConditionTimeout
```

test this please

**Comment by chibenwa** [comment]
```
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR] Failures: 
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR]   ManageableMailQueueContract.getSizeShouldReturnMessageCountWhenSeveralMails:71 expected:<[2]L> but was:<[3]L>
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR] Errors: 
[a25cf2e2ebdf190ddde777be051238429321930e] [ERROR]   MailQueueContract.concurrentEnqueueDequeueShouldNotFail:387 » NotTerminated
```

test this please

**Comment by chibenwa** [comment]
`SwiftTempAuthObjectStorageBlobsDAOBuilderTest.builtBlobsDAOCanStoreAndRetrieve:94 » Timeout`

test this please

**Comment by hoangdat** [comment]
Build triggered. sha1 is merged.

**Comment by hoangdat** [comment]
Build started sha1 is merged.

**Comment by hoangdat** [comment]
Build finished. No test results found.

**Comment by hoangdat** [comment]
Build triggered. sha1 is merged.

**Comment by hoangdat** [comment]
Build started sha1 is merged.

File: src/site/markdown/server/install/guice-cassandra-rabbitmq-swift.md [hunk 2]
```diff
@@ -59,4 +59,13 @@ $ java -Dworking.directory=. -jar target/james-server-cassandra-rabbitmq-guice.j
 ```
 
 ## Guice-cassandra-rabbitmq-ldap
-### Note: this product is not supported yet
+
+You can follow the same guide to build and run guice-cassandra-rabbitmq-swift-ldap artifact, except that:
+ - The **jar** and **libs** needs to be retrieve from server/container/guice/cassandra-rabbitmq-ldap-guice/target after compilation
+ - The sample configuration can be found in [dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf](https://github.com/apache/james-project/tree/master/dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf)
+ - You need a running LDAP server to connect to.
```
Comment (rouazana) [inline src/site/markdown/server/install/guice-cassandra-rabbitmq-swift.md]: maybe you should explain in which file you need to configure this

Comment (hoangdat) [reply src/site/markdown/server/install/guice-cassandra-rabbitmq-swift.md]: I updated the `userrepository.xml` configuration file path to this document.

**Comment by chibenwa** [comment]
`SwiftTempAuthObjectStorageBlobsDAOBuilderTest.builtBlobsDAOCanStoreAndRetrieve:94 » Timeout`

Fixed by https://github.com/linagora/james-project/pull/1967

test this please

**Comment by chibenwa** [comment]
`MailQueueContract.concurrentEnqueueDequeueShouldNotFail:387 » NotTerminated`

test this please

**Comment by chibenwa** [comment]
`MailQueueContract.concurrentEnqueueDequeueShouldNotFail:387 » NotTerminated`

test this please

**Comment by chibenwa** [comment]
`MailQueueContract.concurrentEnqueueDequeueShouldNotFail:387 » NotTerminated`

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**